### PR TITLE
python3Packages.ge25519: init at 0.2.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python implementation of Ed25519 group elements and operations

https://github.com/nthparty/ge25519

This is a dependency of the latest `aiocoap` release. For easier processing the update of `aiocoap` is splitted. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114980` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>python37Packages.bitlist</li>
    <li>python37Packages.fe25519</li>
    <li>python37Packages.fountains</li>
    <li>python37Packages.ge25519</li>
    <li>python37Packages.parts</li>
    <li>python38Packages.bitlist</li>
    <li>python38Packages.fe25519</li>
    <li>python38Packages.fountains</li>
    <li>python38Packages.ge25519</li>
    <li>python38Packages.parts</li>
    <li>python39Packages.bitlist</li>
    <li>python39Packages.fe25519</li>
    <li>python39Packages.fountains</li>
    <li>python39Packages.ge25519</li>
    <li>python39Packages.parts</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114980` at 771840fb run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/22011d4edf5c9edcb8aa66cc2d489971">python37Packages.bitlist</a></li></li>
    <li><a href="https://gist.github.com/5a1c2b17acb6125763d9974373ebd632">python37Packages.fe25519</a></li></li>
    <li><a href="https://gist.github.com/b4f9c2758772ca0f0cd585fb0c39b427">python37Packages.fountains</a></li></li>
    <li><a href="https://gist.github.com/23351366cb6f417ac1868e3573d980e3">python37Packages.ge25519</a></li></li>
    <li><a href="https://gist.github.com/375e4911933d68995542b5a9391d281d">python37Packages.parts</a></li></li>
    <li><a href="https://gist.github.com/ab57adf6e2d54b7e0b5b993ceecc6986">python38Packages.bitlist</a></li></li>
    <li><a href="https://gist.github.com/fe919064759bb3c1892e6e1184a86c06">python38Packages.fe25519</a></li></li>
    <li><a href="https://gist.github.com/ca4851d5d3f2da361247fbd39b4cb9fd">python38Packages.fountains</a></li></li>
    <li><a href="https://gist.github.com/7a17c7439a04871bf6bfac29bc096e07">python38Packages.ge25519</a></li></li>
    <li><a href="https://gist.github.com/fc3a338c2aff561f0a3824231d88261a">python38Packages.parts</a></li></li>
    <li><a href="https://gist.github.com/c3bc88ab492a8684c11d52617b33ad41">python39Packages.bitlist</a></li></li>
    <li><a href="https://gist.github.com/56e330446ad6f34ff43ad272c1602691">python39Packages.fe25519</a></li></li>
    <li><a href="https://gist.github.com/ff3b980c91a4ad4956a2d9e3b7dc8a7d">python39Packages.fountains</a></li></li>
    <li><a href="https://gist.github.com/f7c1423582858a0a1536b6f19a2d6673">python39Packages.ge25519</a></li></li>
    <li><a href="https://gist.github.com/feb95145b16d993998ce87e8304c0be9">python39Packages.parts</a></li></li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114980` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>python37Packages.bitlist</li>
    <li>python37Packages.fe25519</li>
    <li>python37Packages.fountains</li>
    <li>python37Packages.ge25519</li>
    <li>python37Packages.parts</li>
    <li>python38Packages.bitlist</li>
    <li>python38Packages.fe25519</li>
    <li>python38Packages.fountains</li>
    <li>python38Packages.ge25519</li>
    <li>python38Packages.parts</li>
    <li>python39Packages.bitlist</li>
    <li>python39Packages.fe25519</li>
    <li>python39Packages.fountains</li>
    <li>python39Packages.ge25519</li>
    <li>python39Packages.parts</li>
  </ul>
</details>

File: pkgs/development/python-modules/fountains/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    bitlist
+  ];
+
+  # There are no tests at the moment only the requirement in tests_require
+  postPatch = ''
+    sed -i "/tests_require=["nose"],/d" setup.py
+  '';
+
+  doCheck = false;
+  pythonImportsCheck = [ "parts" ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/fountains/default.nix:26]:
```suggestion
  pythonImportsCheck = [ "fountains" ];
```

File: pkgs/development/python-modules/fountains/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = with lib; {
+    description = "Python library for common list functions related to partitioning lists";
+    homepage = "https://github.com/lapets/parts";
```
Comment (dotlambda) [inline pkgs/development/python-modules/fountains/default.nix:30]:
```suggestion
    homepage = "https://github.com/reity/fountains";
```

File: pkgs/development/python-modules/fountains/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  propagatedBuildInputs = [
+    bitlist
+  ];
+
+  # There are no tests at the moment only the requirement in tests_require
+  postPatch = ''
+    sed -i "/tests_require=["nose"],/d" setup.py
+  '';
```
Comment (dotlambda) [inline pkgs/development/python-modules/fountains/default.nix:23]: Why patch this if we set `doCheck = false` anyway?

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/fountains/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "0jk5y099g6ggaq5lwp0jlg4asyhcdxnl3him3ibmzc1k9nnknp30";
+  };
+
+  propagatedBuildInputs = [
+    bitlist
+  ];
+
+  # There are no tests at the moment only the requirement in tests_require
+  postPatch = ''
+    sed -i "/tests_require=["nose"],/d" setup.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fountains/default.nix:22]: Please use substituteInPlace.

File: pkgs/development/python-modules/parts/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  version = "1.0.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1ym238hxwsw15ivvf6gzmkmla08b9hwhdyc3v6rs55wga9j3a4db";
+  };
+
+  # There are no tests at the moment only the requirement in tests_require
+  postPatch = ''
+    sed -i "/tests_require=["nose"],/d" setup.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/parts/default.nix:17]: Please use substituteInPlace.

File: pkgs/development/python-modules/fountains/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/fountains/default.nix:26]: Thanks, update.

File: pkgs/development/python-modules/fountains/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/fountains/default.nix:30]: Thanks

File: pkgs/development/python-modules/fountains/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/fountains/default.nix:23]: I guess that the same approach as for the coverage removal is unnecessary. Will leave `nose` alone.

File: pkgs/development/python-modules/fountains/default.nix (hunk 4 above)
Comment (fabaff) [reply pkgs/development/python-modules/fountains/default.nix:22]: See above. Thanks for the feedback.

File: pkgs/development/python-modules/parts/default.nix (hunk 5 above)
Comment (fabaff) [reply pkgs/development/python-modules/parts/default.nix]: I will leave `nose` alone.

**Comment by fabaff** [comment]
@GrahamcOfBorg build python3Packages.bitlist python3Packages.fe25519 python3Packages.ge25519 python3Packages.fountains python3Packages.parts

**Pull request merged**